File: wear/src/main/java/com/pdyjak/powerampwear/music_browser/categories/CategoryViewHolder.java
package com.pdyjak.powerampwear.music_browser.categories;

import android.support.annotation.NonNull;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.pdyjak.powerampwear.R;
import com.pdyjak.powerampwear.music_browser.ItemViewHolder;

class CategoryViewHolder extends ItemViewHolder {

    @NonNull
    private final ImageView mIcon;
    @NonNull
    private final TextView mName;
    private CategoryItem mCurrent;

    CategoryViewHolder(View itemView) {
        super(itemView);
        mIcon = (ImageView) itemView.findViewById(R.id.icon);
        mName = (TextView) itemView.findViewById(R.id.name);
        itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (mCurrent == null) throw new IllegalStateException("Should never happen");
                mCurrent.onClicked();
            }
        });
    }

    @Override
    public void bindWith(@NonNull Object object) {
        mCurrent = (CategoryItem) object;
        mIcon.setImageResource(mCurrent.iconId);
        mName.setText(itemView.getContext().getText(mCurrent.stringId));
    }
}

File: wear/src/main/java/com/pdyjak/powerampwear/music_browser/categories/CategoryViewHolder.kt
package com.pdyjak.powerampwear.music_browser.categories

import android.view.View
import android.widget.ImageView
import android.widget.TextView

import com.pdyjak.powerampwear.R
import com.pdyjak.powerampwear.music_browser.ItemViewHolder

internal class CategoryViewHolder(itemView: View) : ItemViewHolder(itemView) {

    private val mIcon: ImageView = itemView.findViewById(R.id.icon) as ImageView
    private val mName: TextView = itemView.findViewById(R.id.name) as TextView

    override fun bindWith(obj: Any?) {
        (obj as CategoryItem).let { item ->
            itemView.setOnClickListener { item.onClicked() }
            mIcon.setImageResource(item.iconId)
            mName.text = itemView.context.getText(item.stringId)
        }
    }
}
